velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -121,6 +121,40 @@ class DateTimeFunctionsTest : public functions::test::FunctionBaseTest {
         unpackZoneKeyId(timestampWithTimezone)};
   }
 
+  std::optional<TimestampWithTimezone> timestampWithTimezoneAtTimezone(
+      std::optional<int64_t> timestamp,
+      const std::optional<std::string>& timeZoneName,
+      const std::optional<std::string>& newTimeZoneName) {
+    facebook::velox::VectorPtr resultVector;
+    if (!timestamp.has_value() || !timeZoneName.has_value()) {
```

Comment:
why can't we use `evaluateOnce()` like we do for the other functions?

Reply:
Using the uncasted vector returned from `evaluate()` seemed the best way to simulate returning a `TimestampWithTimezone` "tuple". I think `evaluateOnce()` can only return primitive types.

Reply:
can't you do something like? `evaluateOnce<TimestampWithTimezone>()`

I haven't tried it myself, but I suppose this should work.

Reply:
Tested that, it doesn't work. I based this implementation off `parseDatetime()`, the only other function in here to return a TimestampWithTimezone - so this likely is a limitation with `evaluateOnce()`.